src/lib/router.ts

```diff
@@ -85,12 +85,38 @@ type PaginatedMatches = {
   results: Match[];
 };
 
+type FilterProps = { gameId?: number; username?: string };
+
+function getFilter(filterOptions: FilterProps) {
+  const { gameId, username } = filterOptions;
+  const filter: FilterProps = {};
+
+  if (typeof gameId !== 'undefined') {
+    filter.gameId = Number(gameId);
+  }
+  if (typeof username !== 'undefined') {
+    filter.username = String(username);
+  }
```

Comment:
What happens if the values are arrays?
```suggestion
  if (typeof gameId !== 'undefined' && !Array.isArray(gameId)) {
    filter.gameId = Number(gameId);
  }
  if (typeof username !== 'undefined' && !Array.isArray(username)) {
    filter.username = String(username);
  }
```